ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -224,12 +230,44 @@ public String process(final Expression expression) {
   }
 
   private String formatExpression(final Expression expression) {
+    final SqlToJavaTypeContext context = new SqlToJavaTypeContext();
     final Pair<String, SqlType> expressionFormatterResult =
-        new Formatter(functionRegistry).process(expression, null);
+        new Formatter(functionRegistry).process(expression, context);
     return expressionFormatterResult.getLeft();
   }
 
-  private class Formatter implements ExpressionVisitor<Pair<String, SqlType>, Void> {
+  private static class SqlToJavaTypeContext {
```

Comment:
The context currently between ExpressionTypeManager/SqlToJavaVisitor/CodeGenRunner does the exact same thing in each respective class, can we just have a singular context for now that can be used across all 3? There's nothing really differentiating the contexts currently so I don't see why we'd need to have a specific context class for each

Reply:
I think it can definitely be shared between `SqlToJavaVisitor` and `CodeGenRunner`. The context in `ExpressionTypeManager` was heavily relied on before adding the others though and has some extra methods to resolve sql types so I lean towards leaving that one separate, WDYT?